### ISSUE TITLE
Hover / goto definition on a `new Foo` should show / go to the constructor

### ISSUE THREAD

**Issue by Taluu** [phpactor/phpactor#756]
Currently, when doing a `hover` or a `go to definition` (in vim, plugin OR lsp), we are jumping onto the class definition (`class Foo`).

It would be awesome to be able to jump / show the hover for the constructor (not instead, maybe through the quick fix list, as is done with LSP + intellisense ?)

**Comment by Taluu** [comment]
Hover (on `new ArrayCollection`) :
![image](https://user-images.githubusercontent.com/239685/56959439-a962f780-6b4d-11e9-9244-3ffe5836cd01.png)

Go to definition (on `new Array Collection`): 
![image](https://user-images.githubusercontent.com/239685/56959482-cf889780-6b4d-11e9-93ef-69140071db3b.png)

**Comment by Taluu** [comment]
Actually, making a `hover` in phpactor prints the signature of the constructor or the method linked. So wondering if this is really necessary (well, TBH, it's nice to have the description and the whole text instead of just the signature, but it's also nice like that)

**Comment by dantleech** [comment]
So this is just about the goto definition going to the construtor?

**Comment by Taluu** [comment]
Or on multiple places, such as the constructor and the class itself.

**Comment by dantleech** [comment]
So, concretely this ticket would be to create a quick-fix list and add both the class and constructor to it?

**Comment by Taluu** [comment]
Yep

**Comment by elythyr** [comment]
I'm not sure about that.
First, go to the constructor make sense when it's a public constructor.
With static factory method it's a lot less relevant.
With dependency injection we spend a lot less time on constructing objects so I don't see why the constructor should be more important than the class itself with parents and interfaces. But I agree that last point is more a personal preferences than a con.

What really bother me here is the use of the quickfix.
It's a powerfull tool in Vim and using it here might just counter productive.
The main scenario I have in mind is when running tests.
The quickfix is loaded with the error, I use `phpacotr#GotoDefinition` to quickly go somewhere and fix the problem.
Then my quicklist contains two positions in the same file that can be reach easely through motions.
And I've lost my tests results.

**Comment by Taluu** [comment]
Yes but that is only if you go onto the definition of the class (we could even limit it to when it is instanciated).

Otherwise, keep the current behavior which directly goes to the right method / property.
